wagtailmodelchoosers/rich_text.py

```diff
@@ -0,0 +1,122 @@
+import re
+
+from django.contrib.staticfiles.templatetags.staticfiles import static
+from django.utils.html import format_html, format_html_join
+from draftjs_exporter.dom import DOM
+from wagtail.admin.rich_text.converters.html_to_contentstate import InlineEntityElementHandler
+from wagtail.admin.rich_text.editors.draftail import features as draftail_features
+
+from .utils import get_chooser_options
+
+
+def normalize_entity_name(name):
+    """
+    Lower and replace non-alphanumeric characters with underscores.
+    """
+
+    return re.sub(r'[^\w]', '_', name.lower())
+
+
+def get_chooser_generic_to_database_format(feature_name):
+    entity_name = normalize_entity_name(feature_name)
+    attrname = 'data-{}'.format(entity_name.replace('_', '-'))
+
+    def entity_decorator(props):
+        return DOM.create_element(
+            'span',
+            {attrname: props['value']},
+            props['children'],
+        )
+    entity_decorator.__name__ = '{}_entity_decorator'.format(feature_name)
+
+    return entity_decorator
+
+
+def get_chooser_generic_from_database_format(feature_name):
+    entity_name = normalize_entity_name(feature_name)
+    attrname = 'data-{}'.format(entity_name.replace('_', '-'))
+    klassname = '{}EntityElementHandler'.format(entity_name.title().replace('_', ''))
+
+    class BaseEntityElementHandler(InlineEntityElementHandler):
+        mutability = 'IMMUTABLE'
```

Comment:
Should this support both mutable and immutable entities like the client-side code does? I might be missing something.